lib/notifications/display.dart

```diff
@@ -94,34 +95,48 @@ class NotificationChannelManager {
 
 /// Service for managing the notifications shown to the user.
 class NotificationDisplayManager {
+  /// Reset the state of the [NotificationDisplayManager], for testing.
+  static void debugReset() {
+    _backgroundIsolateGlobalStore = null;
+  }
+
   static Future<void> init() async {
-    await ZulipBinding.instance.notifications.initialize(
-      const InitializationSettings(
-        android: AndroidInitializationSettings('zulip_notification'),
-      ),
-      onDidReceiveNotificationResponse: _onNotificationOpened,
-    );
-    final launchDetails = await ZulipBinding.instance.notifications.getNotificationAppLaunchDetails();
-    if (launchDetails?.didNotificationLaunchApp ?? false) {
-      _handleNotificationAppLaunch(launchDetails!.notificationResponse);
-    }
     await NotificationChannelManager.ensureChannel();
   }
 
-  static void onFcmMessage(FcmMessage data, Map<String, dynamic> dataJson) {
+  static void onFcmMessage(FcmMessage data,Map<String, dynamic> dataJson, {
+    required bool isBackground,
+  }) {
     switch (data) {
-      case MessageFcmMessage(): _onMessageFcmMessage(data, dataJson);
+      case MessageFcmMessage(): _onMessageFcmMessage(data, dataJson, isBackground: isBackground);
       case RemoveFcmMessage(): _onRemoveFcmMessage(data);
       case UnexpectedFcmMessage(): break; // TODO(log)
     }
   }
 
-  static Future<void> _onMessageFcmMessage(MessageFcmMessage data, Map<String, dynamic> dataJson) async {
+  static Future<void> _onMessageFcmMessage(MessageFcmMessage data, Map<String, dynamic> dataJson, {
+    required bool isBackground,
+  }) async {
     assert(debugLog('notif message content: ${data.content}'));
+
+    final GlobalStore globalStore;
+    if (isBackground) {
+      _backgroundIsolateGlobalStore ??= await ZulipBinding.instance.loadGlobalStore();
+      globalStore = _backgroundIsolateGlobalStore!;
+    } else {
+      NavigatorState navigator = await ZulipApp.navigator;
+      final context = navigator.context;
+      assert(context.mounted);
+      if (!context.mounted) return; // TODO(linter): this is impossible as there's no actual async gap, but the use_build_context_synchronously lint doesn't see that
+      globalStore = GlobalStoreWidget.of(context);
+    }
```

Comment:
Hmm, it feels like the wrong structure for this NotificationDisplayManager class to be maintaining its own GlobalStore instance (in _backgroundIsolateGlobalStore).

Having this method, and its caller chain, need to know about whether it's running in a background isolate or in the foreground seems messy too.

I've just sent #1005 to solve this a different way. Then this becomes:
```suggestion
    final globalStore = await ZulipBinding.instance.getGlobalStore();
```